### PR TITLE
Add module for AWS IAM Identity Federation (SAML) management

### PR DESCRIPTION
**Pull request by aidan-**
##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
cloud/amazon/iam_saml_federation.py

##### ANSIBLE VERSION
```
ansible 2.2.0.0
```

##### SUMMARY
Adds support for managing AWS IAM Identity Federation (SAML) providers.
Supports check_mode.

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @aidan- for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

File: cloud/amazon/iam_saml_federation.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+EXAMPLES = '''
+# Note: None of these examples set aws_access_key or aws_secret_key.
+# It is assumed that their matching environment variables are set.
+
+# Creates a new iam saml identity provider if not present
+- name: saml provider
+  iam_saml_federation:
+    name: example1
+    saml_metadata_document: >
```
Comment (ryansb) [inline cloud/amazon/iam_saml_federation.py]: Can you also put a comment in here about what the `>` means when opening a text block? Something to the effect of:
```
# the > here opens an indented block, so no escaping/quoting is needed when in the indentation level under this key
```

File: cloud/amazon/iam_saml_federation.py [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+          except botocore.exceptions.ClientError as e:
+            res['msg'] = str(e)
+            res['debug'] = [arn, metadata]
+            self.module.fail_json(**res)
+
+    else: #create
+      res['changed'] = True
+      if not self.module.check_mode:
+        try:
+          resp = self.conn.create_saml_provider(SAMLMetadataDocument=metadata, Name=name)
```
Comment (ryansb) [inline cloud/amazon/iam_saml_federation.py]: Can you return the provider ARN in both the update & create cases? `boto3` returns them as `SAMLProviderArn` on the response object. http://boto3.readthedocs.io/en/latest/reference/services/iam.html#IAM.Client.create_saml_provider

File: cloud/amazon/iam_saml_federation.py [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+description:
+    - Maintains IAM SAML identity federation configurations.  This module has a dependency on python-boto >= 2.5.
+options:
+  name:
+    description:
+      - The name of the provider to create.
+    required: true
+  saml_metadata_document:
+   description:
+      - An  XML  document  generated by an identity provider (IdP) that supports SAML 2.0.
```
Comment (ryansb) [inline cloud/amazon/iam_saml_federation.py]: Can you link to the AWS docs that explain this XML doc? The only reference I can find is docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_saml.html

**Review by ryansb — CHANGES_REQUESTED** [review]

**Comment by ryansb** [comment]
Also, it appears this file is indented with 2-space tabs. The Ansible style guides follow the PEP8-standard 4-space indentation. Can you please switch to that style?

**Comment by aidan-** [comment]
Thank you for the feedback @ryansb, those issues have now been fixed.

**Pull request closed without merging**